pkg/sandbox/infra.go

```diff
@@ -0,0 +1,67 @@
+package sandbox
+
+import (
+	"fmt"
+
+	"github.com/cri-o/cri-o/internal/storage"
+	libconfig "github.com/cri-o/cri-o/pkg/config"
+	"github.com/cri-o/cri-o/pkg/container"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/opencontainers/runtime-tools/generate"
+)
+
+func (s *sandbox) InitializeInfraContainer(serverConfig *libconfig.Config, podContainer *storage.ContainerInfo) error {
+	var err error
+	s.infra, err = container.New()
+	if err != nil {
+		return err
+	}
+	g := s.infra.Spec()
+	g.HostSpecific = true
+	g.ClearProcessRlimits()
+
+	// setup defaults for the pod sandbox
+	g.SetRootReadonly(true)
+
+	// configure default ulimits
+	for _, u := range serverConfig.Ulimits() {
+		g.AddProcessRlimits(u.Name, u.Hard, u.Soft)
+	}
+
+	pauseCommand, err := PauseCommand(serverConfig, podContainer.Config)
+	if err != nil {
+		return err
+	}
+	g.SetProcessArgs(pauseCommand)
+
+	return nil
+}
+
+func (s *sandbox) Spec() *generate.Generator {
+	return s.infra.Spec()
```

Comment:
this might panic if called before `InitializeInfraContainer`.

Reply:
I don't think we should gracefully recover from that situation, that feels like a good time for an assert. Would you like the panic to be more deliberate?

Reply:
documentation could use updating too

Reply:
I updated the docs of the function to make clear the required ordering